lib/src/view/strand_color_picker.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:color/color.dart';
+import 'package:over_react/over_react.dart';
+import 'package:over_react/over_react_redux.dart';
+import 'package:scadnano/src/actions/actions.dart' as actions;
+import 'package:scadnano/src/view/react_color.dart';
+
+import '../app.dart';
+import '../state/app_state.dart';
+import '../state/strand.dart';
+
+part 'strand_color_picker.over_react.g.dart';
+
+typedef ActionCreator = actions.UndoableAction Function(Strand strand);
+
+UiFactory<StrandColorPickerProps> ConnectedStrandColorPicker =
+    connect<AppState, StrandColorPickerProps>(mapStateToProps: (state) {
+  return StrandColorPicker()
+    ..color = state.ui_state.strand_color_picker_strand?.color
+    ..show = state.ui_state.strand_color_picker_strand != null
+    ..strand = state.ui_state.strand_color_picker_strand;
+})(StrandColorPicker);
+
+UiFactory<StrandColorPickerProps> StrandColorPicker = _$StrandColorPicker;
+
+mixin StrandColorPickerProps on UiProps {
+  Color color;
+  bool show;
+  Strand strand;
+}
+
+@State()
+mixin StrandColorPickerState on UiState {
+  Color color;
+}
+
+class StrandColorPickerComponent
+    extends UiStatefulComponent2<StrandColorPickerProps, StrandColorPickerState> {
+  void handleOnChangeComplete(color, event) {
+    state.color = HexColor(color.hex);
+  }
+
+  void handleOnCancel(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    app.dispatch(actions.StrandColorPickerHide());
+  }
+
+  void handleOnOK(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    app.dispatch(actions.StrandColorPickerHide());
+    Color color = state.color;
+    if (color != null && color != props.strand.color) {
+      actions.Action action = batch_if_multiple_selected(color_set_strand_action_creator(color), props.strand,
+          app.state.ui_state.selectables_store.selected_strands);
+      app.dispatch(action);
+    }
+  }
+
+  @override
+  render() {
+    if (props.show) {
+      return (Dom.div()..className = 'dialog-form-container')(
+        (Dom.div()..className = 'dialog-form')(
+          (Dom.form()
+            ..onSubmit = handleOnOK
+            ..className = 'dialog-form-form')(
+            ChromePicker({
```

Comment:
Using the React Color component produces some React deprecation warnings in the console. Would be nice to figure out how to fix/hide them

Reply:
Duplicate: was trying to change the selected code in the Github comment